File: app/src/main/java/de/wollis_page/gibsonos/dto/desktop/Item.java
package de.wollis_page.gibsonos.dto.desktop;

import androidx.annotation.NonNull;

public class Item
{
    @NonNull
    private String module;
    @NonNull
    private String task;
    @NonNull
    private String action;
    private String text;
    private String icon;
    private String thumb;
    private String customIcon;
    //public final JSONObject params;

    public Item(
        @NonNull String module,
        @NonNull String task,
        @NonNull String action,
        String text,
        String icon,
        String thumb,
        String customIcon//,
        //JSONObject params
    ) {
        this.module = module;
        this.task = task;
        this.action = action;
        this.text = text;
        this.icon = icon;
        this.thumb = thumb;
        this.customIcon = customIcon;
        //this.params = params;
    }

    @NonNull
    public String getModule() {
        return module;
    }

    public void setModule(@NonNull String module) {
        this.module = module;
    }

    @NonNull
    public String getTask() {
        return task;
    }

    public void setTask(@NonNull String task) {
        this.task = task;
    }

    @NonNull
    public String getAction() {
        return action;
    }

    public void setAction(@NonNull String action) {
        this.action = action;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getIcon() {
        return icon;
    }

    public void setIcon(String icon) {
        this.icon = icon;
    }

    public String getThumb() {
        return thumb;
    }

    public void setThumb(String thumb) {
        this.thumb = thumb;
    }

    public String getCustomIcon() {
        return customIcon;
    }

    public void setCustomIcon(String customIcon) {
        this.customIcon = customIcon;
    }
}

File: app/src/main/java/de/wollis_page/gibsonos/module/core/desktop/dto/Item.kt
package de.wollis_page.gibsonos.module.core.desktop.dto

import android.os.Parcel
import android.os.Parcelable
import android.util.ArrayMap
import de.wollis_page.gibsonos.dto.ListItemInterface

class Item(
    var module: String,
    var task: String,
    var action: String,
    var text: String,
    var icon: String,
    var thumb: String,
    var customIcon: Long,
    var params: Map<String, Any>?
): ListItemInterface, Parcelable {
    constructor(parcel: Parcel) : this(
        parcel.readString().toString(),
        parcel.readString().toString(),
        parcel.readString().toString(),
        parcel.readString().toString(),
        parcel.readString().toString(),
        parcel.readString().toString(),
        parcel.readLong(),
        null
    ) {
        this.params = this.readParamsFromParcel(parcel)
    }

    private fun readParamsFromParcel(parcel: Parcel): Map<String, Any> {
        val params = ArrayMap<String, Any>()
        parcel.readMap(params, Map::class.java.classLoader)

        return params
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(this.module)
        parcel.writeString(this.task)
        parcel.writeString(this.action)
        parcel.writeString(this.text)
        parcel.writeString(this.icon)
        parcel.writeString(this.thumb)
        parcel.writeLong(this.customIcon)
        parcel.writeMap(this.params)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Item> {
        override fun createFromParcel(parcel: Parcel): Item {
            return Item(parcel)
        }

        override fun newArray(size: Int): Array<Item?> {
            return arrayOfNulls(size)
        }
    }
}